docs/aad-application-configuration.md

```diff
@@ -0,0 +1,115 @@
+
+# AAD application configuration
+
+The deployed OpenShift cluster needs a valid AAD application and service principal to call back into the Azure API,
+in order to enable AAD authentication. There are a few options here:
+
+* Create application using Azure Portal and configure all values there.
+
+* Create application using CLI and later configure permission in the Azure portal.
+
+* Create application using CLI and approve permission in the Azure portal
+
+To specify your own application for OpenShift to use, add AAD flags to the `az openshift create` command:
+
+```azurecli-interactive
+OSA_CLUSTER_NAME=mjudeikis-prod1
+LOCATION=eastus
+OSA_FQDN=$OSA_CLUSTER_NAME.$LOCATION.cloudapp.azure.com
+
+az openshift create --resource-group $OSA_CLUSTER_NAME --name $OSA_CLUSTER_NAME -l $LOCATION --node-count 4 --fqdn $OSA_FQDN --aad-client-app-id $OSA_AAD_ID --aad-client-app-secret $OSA_AAD_SECRET --aad-tenant-id $OSA_AAD_TENANT
+```
+
+### Option 1: Create application using Azure portal
+
+Do a search for `App registrations` in the search section located at the top and navigate to it or go to AAD -> app registrations
+
+![](./medias/OSA_APP_Portal.png)
+
+Press `New application registration` button. Enter details for your application instance and press `Create`.
+
+![](./medias/OSA_AAD_Portal_Create.png)
+
+ Sign-on URL is OpenShift web console URL: `https://awesome-cluster.eastus.cloudapp.azure.com/oauth2callback/Azure%20AD`
+
+Copy value of the `Application ID` fields for `OSA_AAD_ID` from [step 4 above](#Step-4:-Create-OpenShift-cluster). 
+
+![](./medias/OSA_AAD_Portal_Details.png)
+
+`OSA_AAD_SECRET` Value should be set to `Key`, attached to the AAD application. Under same AAD application press `Keys` and create new key:
+
+![](./medias/OSA_AAD_Portal_key_create.png)
+
+> Once this key is created and the page is left the value will be hidden. Make sure to store is somewhere safe.
+
+Key value is used for `az openshift create` command `--aad-client-app-secret <aad key value>`
+
+Next you will need to grant set of API permissions for sign-on to work. Under same application press `Required permissions` and `Add`. Set these values:
+
+```
+Windows Azure Active Directory:
```

Comment:
Missing the Graph API permissions now or is it expected to be added by the cli during create?

Reply:
as @amanohar stated we dont need them for PP

Reply:
missed that comment, thanks.  Please make sure it's tested :smile:

Reply:
did one this morning. Worked like a charm. The main difference is that with Graph api you get a seamless redirect and with this one user have to "approve it once in azure portal"